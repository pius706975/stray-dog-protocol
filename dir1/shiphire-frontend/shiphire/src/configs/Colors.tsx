export const Color = {
    bgColor: "#f0f6fb",
    softSecBgPrimary: "#bcd3e6",
    softGreyBgPrimary: "#cbcdd1",
    primaryColor: "#216dab",
    softPrimaryColor: "#6499c4",
    secColor: "#9cd2ff",
    softSecColor: "#d7edff",
    darkTextColor: "#4d5963",
    lightTextColor: "#f0f6fb",
    bgSuccessColor: "#e9faec",
    boldSuccessColor: "#1fcc44",
    successColor: "#4cd669",
    bgErrorColor: "#fae9e7",
    boldErrorColor: "#cc1f12",
    errorColor: "#d64c41",
    bgInfoColor: "#d0dbf5",
    boldInfoColor: "#124ccc",
    infoColor: "#5982db",
    bgWarningColor: "#fbf6e8",
    boldWarningColor: "#d9a318",
    warningColor: "#e1b546",
    bgNeutralColor: "#dbdee0",
    neutralColor: "#828b92",
    primaryDisableColor: "#6499c4"
}