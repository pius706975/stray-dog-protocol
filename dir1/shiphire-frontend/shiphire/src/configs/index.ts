export * from "./Fonts"
export * from "./Colors"
export * from "./LocalStorage"
export * from "./Svg"
export * from "./StatusText"
export * from "./firebase/GoogleSignIn"
export * from "./firebase/AppleSignIn"