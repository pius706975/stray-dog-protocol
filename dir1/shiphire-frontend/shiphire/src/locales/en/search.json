{
    "searchFieldPlaceholder": "Search ship name or company",
    "emptyList": "No ships or companies found.",
    "filterModal": {
        "title": "Ship Categories",
        "location": "Ship Location",
        "filterSubtitle": "Rent Date Availability",
        "startDateLabel": "Select Start Date",
        "startDatePlaceholder": "Select End Date",
        "endDateLabel": "Select Start Date",
        "endDatePlaceholder": "Select End Date",
        "provincePlaceholder": "Select province",
        "cityPlaceholder": "Select city",
        "searchPlaceholder": "Search...",
        "btnClose": "Close",
        "btnApply": "Apply Filter"
    },
    "rentCount": "Rented"
}
