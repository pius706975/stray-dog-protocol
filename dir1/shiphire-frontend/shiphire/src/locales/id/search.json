{
    "searchFieldPlaceholder": "Cari nama kapal atau perusahaan",
    "emptyList": "Tidak ada kapal atau perusahaan yang ditemukan",
    "filterModal": {
        "title": "Kategori Kapal",
        "location": "Lokasi Kapal",
        "filterSubtitle": "Tanggal Ketersediaan Sewa",
        "startDateLabel": "Pilih Tanggal Mulai",
        "startDatePlaceholder": "Pilih Tanggal Mulai",
        "endDateLabel": "Pilih Tanggal Akhir",
        "endDatePlaceholder": "Pilih Tanggal Akhir",
        "provincePlaceholder": "Pilih provinsi",
        "cityPlaceholder": "Pilih Kota",
        "searchPlaceholder": "Cari...",
        "btnClose": "Tutup",
        "btnApply": "Terapkan Filter"
    },
    "rentCount": "Disewa"
}
