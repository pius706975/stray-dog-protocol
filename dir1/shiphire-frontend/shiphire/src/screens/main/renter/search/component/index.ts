export { default as DatePickerFilter } from './DatePickerFilter';
