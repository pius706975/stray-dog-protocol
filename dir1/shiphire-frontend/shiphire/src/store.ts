import { configureStore } from '@reduxjs/toolkit';
import {
    addShipSlice,
    documentNameSlice,
    modalSlice,
    networkErrorSlice,
    notifBadgeSlice,
    progressIndicatorSlice,
    renterPreferenceSlice,
    requestForQuoteSlice,
    shipSpecSlice,
    termsStatusSlice,
    userLanguageSlice,
    userLocationSlice,
    userStatusSlice,
} from './slices';

const store = configureStore({
    reducer: {
        modal: modalSlice.reducer,
        progressIndicator: progressIndicatorSlice.reducer,
        renterPreference: renterPreferenceSlice.reducer,
        requestForQuote: requestForQuoteSlice.reducer,
        userStatus: userStatusSlice.reducer,
        addShip: addShipSlice.reducer,
        documentName: documentNameSlice.reducer,
        termsStatus: termsStatusSlice.reducer,
        userLanguage: userLanguageSlice.reducer,
        userLocation: userLocationSlice.reducer,
        shipSpecSlice: shipSpecSlice.reducer,
        notifBadgeSlice: notifBadgeSlice.reducer,
        networkError: networkErrorSlice.reducer,
    },
});

export default store;
