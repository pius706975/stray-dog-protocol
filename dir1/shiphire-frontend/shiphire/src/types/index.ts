export * from './Components'
export * from './Navigators'
export * from './ReduxToolkit'
export * from './Screens'
export * from './Services'