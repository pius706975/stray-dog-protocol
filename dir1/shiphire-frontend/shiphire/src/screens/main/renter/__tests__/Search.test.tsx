import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { QueryClient, QueryClientProvider } from 'react-query';
import { MainStackParamList } from '../../../../types';
import MockAdapter from 'axios-mock-adapter';
import httpRequest from '../../../../services/api';
import { Provider } from 'react-redux';
import store from '../../../../store';
import { NavigationContainer } from '@react-navigation/native';
import { Search } from '../search';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import { act } from 'react-test-renderer';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { TOKEN, USERDATA } from '../../../../configs';

const queryClient = new QueryClient();
const MainStack = createNativeStackNavigator<MainStackParamList>();
const mockAdapter = new MockAdapter(httpRequest);

const mockCategories = {
    status: 'success',
    data: [
        {
            _id: '64c764bbdb67b394e843b139',
            name: 'Barge',
            __v: 0,
        },
        {
            _id: '64c769bddb67b394e843b16c',
            name: 'Tugboat',
            __v: 0,
        },
        {
            _id: '64c76a7ddb67b394e843b1af',
            name: 'Ferry',
            __v: 0,
        },
    ],
};

const mockSearching = {
    status: 'success',
    data: [
        {
            _id: '1',
            name: 'barge',
            categories: 'barge',
            companyName: 'PT. Barge',
            rating: '4.5',
            totalRentalCount: '100',
            imageUrl: 'https://www.google.com',
            pricePerMonth: 9000000,
            shipApproved: [
                {
                    name: 'barge',
                    desc: 'barge',
                    approvedShip: true,
                    _id: '1',
                },
            ],
        },
        {
            _id: '2',
            name: 'coastal',
            categories: 'Ferry',
            companyName: 'PT. Coastal',
            rating: '4.8',
            totalRentalCount: '80',
            imageUrl: 'https://www.google.com',
            pricePerMonth: 12000000,
            shipApproved: [
                {
                    name: 'coastal',
                    desc: 'coastal',
                    approvedShip: true,
                    _id: '1',
                },
            ],
        },
    ],
};

const mockScreen = (
    <QueryClientProvider client={queryClient}>
        <Provider store={store}>
            <NavigationContainer>
                <MainStack.Navigator>
                    <MainStack.Screen name="Search" component={Search} />
                </MainStack.Navigator>
            </NavigationContainer>
        </Provider>
    </QueryClientProvider>
);

mockAdapter.onGet('/ship/get-ship-categories').reply(200, mockCategories);

mockAdapter
    .onGet(
        '/ship/search-ship?searchTerm=&category=&inputRentStartDate=02Jan2023&inputRentEndDate=30Mar2024',
    )
    .reply(200, mockSearching);

AsyncStorage.setItem(
    USERDATA,
    JSON.stringify({
        name: 'Usmanul',
        email: 'usman@email.com',
        password: 'test',
        phoneNumber: '12313123',
        isVerified: false,
        isCompanySubmitted: true,
    }),
);
AsyncStorage.setItem(
    TOKEN,
    JSON.stringify({
        accessToken: 'test',
        refreshToken: 'test',
    }),
);

describe('Testing search screen', () => {
    describe('Snapshot Testing', () => {
        it('should render search screen correctly', async () => {
            const searchScreen = render(mockScreen).toJSON();
            await act(async () => {
                expect(searchScreen).toMatchSnapshot();
            });
        });
    });
    describe('Unit Testing', () => {
        it('should render search screen correctly', async () => {
            const { getByTestId } = render(mockScreen);
            const searchScreen = getByTestId('SearchScreen');

            await act(async () => {
                expect(searchScreen).toBeTruthy();
                expect(getByTestId('shipList')).toBeDefined();
            });
        });
        test('searching input integration', async () => {
            const { getByPlaceholderText, getByTestId } = render(mockScreen);

            const searchInput = getByPlaceholderText('searchFieldPlaceholder');

            await act(async () => {
                fireEvent.changeText(searchInput, 'Ba');
            });

            await waitFor(async () => {
                expect(searchInput).toBeTruthy();
            });

            jest.advanceTimersByTime(800);

            mockAdapter
                .onGet(
                    '/ship/search-ship?searchTerm=Ba&category=&inputRentStartDate=&inputRentEndDate=',
                )
                .reply(200, mockSearching);

            await waitFor(async () => {
                expect(getByTestId('shipList')).toBeDefined();
            });
        });
        test('searching input integration and clicked', async () => {
            const { getByPlaceholderText, getByTestId, findByTestId } =
                render(mockScreen);

            const searchInput = getByPlaceholderText('searchFieldPlaceholder');

            await act(async () => {
                fireEvent.changeText(searchInput, 'Ba');
            });

            await waitFor(async () => {
                expect(searchInput).toBeTruthy();
            });

            jest.advanceTimersByTime(800);

            mockAdapter
                .onGet(
                    '/ship/search-ship?searchTerm=Ba&category=&inputRentStartDate=&inputRentEndDate=',
                )
                .reply(200, mockSearching);

            await waitFor(async () => {
                expect(getByTestId('shipList')).toBeDefined();
            });

            await act(async () => {
                fireEvent.press(getByTestId('shipList'));
            });

            // expect(findByTestId('DetailShipScreen')).toBeTruthy();
        });
        test('modal filter to be close', async () => {
            const { getByTestId } = render(mockScreen);

            const filterButton = getByTestId('filterButton');

            await act(async () => {
                fireEvent.press(filterButton);
            });

            await waitFor(async () => {
                const filterModal = getByTestId('filterModal');
                expect(filterModal).toBeTruthy();
            });

            const closeFilter = getByTestId('btnCloseFilterModal');

            await act(async () => {
                fireEvent.press(closeFilter);
            });

            expect(closeFilter).toBeTruthy();
        });
        test('modal apply filter', async () => {
            const { getByTestId } = render(mockScreen);

            const filterButton = getByTestId('filterButton');

            await act(async () => {
                fireEvent.press(filterButton);
            });

            await waitFor(async () => {
                const filterModal = getByTestId('filterModal');
                expect(filterModal).toBeTruthy();
            });

            const applyFilter = getByTestId('btnApplyFilterModal');

            await act(async () => {
                fireEvent.press(applyFilter);
            });

            expect(applyFilter).toBeTruthy();
        });
        test('searching input', async () => {
            const { getByPlaceholderText } = render(mockScreen);

            const searchInput = getByPlaceholderText('searchFieldPlaceholder');

            await act(async () => {
                fireEvent.changeText(searchInput, 'Ba');
            });

            await waitFor(async () => {
                expect(searchInput).toBeTruthy();
            });
        });
    });
});
describe('integration Error', () => {
    it('should error get API categories', async () => {
        await AsyncStorage.setItem(
            USERDATA,
            JSON.stringify({
                name: 'Usmanul',
                email: 'usman@email.com',
                password: 'test',
                phoneNumber: '12313123',
                isVerified: false,
                isCompanySubmitted: true,
            }),
        );
        await AsyncStorage.setItem(
            TOKEN,
            JSON.stringify({
                accessToken: '',
                refreshToken: '',
            }),
        );
        render(mockScreen);
        mockAdapter
            .onGet('/ship/get-ship-categories')
            .reply(401, { status: 'failed' });
        await waitFor(async () => {
            const modalState = store.getState().modal;
            jest.advanceTimersByTime(4000);
            expect(modalState).toEqual({
                visible: true,
                status: 'failed',
                text: 'Token expired, please re-sign in',
            });
        });
    });
    it('should error get API search', async () => {
        render(mockScreen);
        mockAdapter
            .onGet(
                '/ship/search-ship?searchTerm=&category=&inputRentStartDate=&inputRentEndDate=',
            )
            .reply(500, { status: 'failed' });
    });
});
