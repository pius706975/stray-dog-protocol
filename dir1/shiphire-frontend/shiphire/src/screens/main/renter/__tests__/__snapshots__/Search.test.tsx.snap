// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing search screen Snapshot Testing should render search screen correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <RCTScrollView
            collapsable={false}
            data={
              [
                "",
              ]
            }
            getItem={[Function]}
            getItemCount={[Function]}
            handlerTag={1}
            handlerType="NativeViewGestureHandler"
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            renderScrollComponent={[Function]}
            scrollEventThrottle={50}
            showsVerticalScrollIndicator={false}
            stickyHeaderIndices={[]}
            style={
              {
                "marginHorizontal": 5,
              }
            }
            viewabilityConfigCallbackPairs={[]}
            waitFor={
              [
                {
                  "current": null,
                },
                {
                  "current": null,
                },
              ]
            }
          >
            <View>
              <View
                onFocusCapture={[Function]}
                onLayout={[Function]}
                style={null}
              >
                <View
                  style={
                    [
                      undefined,
                      undefined,
                      {
                        "flex": 1,
                      },
                      undefined,
                      {},
                      {},
                      {},
                      {
                        "backgroundColor": "#f0f6fb",
                        "justifyContent": "space-between",
                        "position": "relative",
                      },
                    ]
                  }
                  testID="SearchScreen"
                >
                  <RCTScrollView
                    contentContainerStyle={
                      {
                        "alignItems": undefined,
                        "flex": undefined,
                        "flexGrow": 1,
                        "gap": undefined,
                        "justifyContent": undefined,
                        "marginBottom": undefined,
                        "padding": 15,
                        "paddingVertical": undefined,
                      }
                    }
                  >
                    <View>
                      <View
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "justifyContent": "space-between",
                              "paddingBottom": 5,
                              "paddingTop": 10,
                            },
                          ]
                        }
                      >
                        <View
                          style={
                            [
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              {
                                "width": "85%",
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              {
                                "backgroundColor": "#6499c4",
                                "borderBottomColor": "#e1e1e1",
                                "borderBottomWidth": 1,
                                "borderRadius": 20,
                                "borderTopColor": "#e1e1e1",
                                "borderTopWidth": 1,
                                "padding": 8,
                              }
                            }
                            testID="RNE__SearchBar-wrapper"
                          >
                            <View
                              style={
                                {
                                  "paddingHorizontal": 0,
                                  "width": "100%",
                                }
                              }
                              testID="RNE__Input__view-wrapper"
                            >
                              <View
                                collapsable={false}
                                style={
                                  {
                                    "alignItems": "center",
                                    "backgroundColor": "white",
                                    "borderBottomWidth": 0,
                                    "borderColor": "#86939e",
                                    "borderRadius": 15,
                                    "flexDirection": "row",
                                    "minHeight": 30,
                                    "overflow": "hidden",
                                    "transform": [
                                      {
                                        "translateX": 0,
                                      },
                                    ],
                                  }
                                }
                              >
                                <TextInput
                                  editable={true}
                                  onBlur={[Function]}
                                  onChangeText={[Function]}
                                  onFocus={[Function]}
                                  placeholder="searchFieldPlaceholder"
                                  placeholderTextColor="#86939e"
                                  platform="default"
                                  style={
                                    {
                                      "color": "#4d5963",
                                      "flex": 1,
                                      "fontFamily": "Kanit-Regular",
                                      "fontSize": 18,
                                      "marginLeft": 10,
                                      "minHeight": 40,
                                    }
                                  }
                                  testID="RNE__SearchBar"
                                  underlineColorAndroid="transparent"
                                  value=""
                                />
                                <View
                                  style={
                                    {
                                      "alignItems": "center",
                                      "height": 40,
                                      "justifyContent": "center",
                                      "marginRight": 8,
                                      "marginVertical": 4,
                                      "paddingRight": 4,
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      {
                                        "flexDirection": "row",
                                      }
                                    }
                                  />
                                </View>
                              </View>
                              <Text
                                style={
                                  {
                                    "color": "#ff190c",
                                    "fontSize": 12,
                                    "height": 0,
                                    "margin": 0,
                                    "padding": 0,
                                  }
                                }
                              />
                            </View>
                          </View>
                        </View>
                        <View
                          style={
                            [
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              {
                                "borderColor": "#6499c4",
                                "borderRadius": 20,
                                "borderWidth": 3,
                                "padding": 15,
                                "paddingBottom": 5,
                                "paddingLeft": 5,
                                "paddingRight": 5,
                                "paddingTop": 5,
                              },
                            ]
                          }
                        >
                          <View
                            accessibilityState={
                              {
                                "busy": undefined,
                                "checked": undefined,
                                "disabled": undefined,
                                "expanded": undefined,
                                "selected": undefined,
                              }
                            }
                            accessibilityValue={
                              {
                                "max": undefined,
                                "min": undefined,
                                "now": undefined,
                                "text": undefined,
                              }
                            }
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            testID="filterButton"
                          >
                            <RNSVGSvgView
                              align="xMidYMid"
                              bbHeight="24"
                              bbWidth="24"
                              fill="none"
                              focusable={false}
                              height={24}
                              meetOrSlice={0}
                              minX={0}
                              minY={0}
                              style={
                                [
                                  {
                                    "backgroundColor": "transparent",
                                    "borderWidth": 0,
                                  },
                                  {
                                    "flex": 0,
                                    "height": 24,
                                    "width": 24,
                                  },
                                ]
                              }
                              vbHeight={24}
                              vbWidth={24}
                              width={24}
                              xml="
    <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<g clip-path="url(#clip0_77_27295)">
<path d="M4.24993 5.61C6.56993 8.59 9.99993 13 9.99993 13V18C9.99993 19.1 10.8999 20 11.9999 20C13.0999 20 13.9999 19.1 13.9999 18V13C13.9999 13 17.4299 8.59 19.7499 5.61C20.2599 4.95 19.7899 4 18.9499 4H5.03993C4.20993 4 3.73993 4.95 4.24993 5.61Z" fill="#323232"/>
</g>
<defs>
<clipPath id="clip0_77_27295">
<rect width="24" height="24" fill="white"/>
</clipPath>
</defs>
</svg>

    
    "
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <RNSVGGroup
                                fill={null}
                                propList={
                                  [
                                    "fill",
                                  ]
                                }
                              >
                                <RNSVGGroup
                                  clipPath="clip0_77_27295"
                                  fill={
                                    {
                                      "payload": 4278190080,
                                      "type": 0,
                                    }
                                  }
                                >
                                  <RNSVGPath
                                    d="M4.24993 5.61C6.56993 8.59 9.99993 13 9.99993 13V18C9.99993 19.1 10.8999 20 11.9999 20C13.0999 20 13.9999 19.1 13.9999 18V13C13.9999 13 17.4299 8.59 19.7499 5.61C20.2599 4.95 19.7899 4 18.9499 4H5.03993C4.20993 4 3.73993 4.95 4.24993 5.61Z"
                                    fill={
                                      {
                                        "payload": 4281479730,
                                        "type": 0,
                                      }
                                    }
                                    propList={
                                      [
                                        "fill",
                                      ]
                                    }
                                  />
                                </RNSVGGroup>
                                <RNSVGDefs>
                                  <RNSVGClipPath
                                    fill={
                                      {
                                        "payload": 4278190080,
                                        "type": 0,
                                      }
                                    }
                                    name="clip0_77_27295"
                                  >
                                    <RNSVGRect
                                      fill={
                                        {
                                          "payload": 4294967295,
                                          "type": 0,
                                        }
                                      }
                                      height="24"
                                      propList={
                                        [
                                          "fill",
                                        ]
                                      }
                                      width="24"
                                      x="0"
                                      y="0"
                                    />
                                  </RNSVGClipPath>
                                </RNSVGDefs>
                              </RNSVGGroup>
                            </RNSVGSvgView>
                          </View>
                        </View>
                      </View>
                      <View
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                          ]
                        }
                        testID="shipList"
                      >
                        <RCTScrollView
                          collapsable={false}
                          contentContainerStyle={
                            {
                              "gap": 14,
                            }
                          }
                          getItem={[Function]}
                          getItemCount={[Function]}
                          handlerTag={2}
                          handlerType="NativeViewGestureHandler"
                          keyExtractor={[Function]}
                          onContentSizeChange={[Function]}
                          onGestureHandlerEvent={[Function]}
                          onGestureHandlerStateChange={[Function]}
                          onLayout={[Function]}
                          onMomentumScrollBegin={[Function]}
                          onMomentumScrollEnd={[Function]}
                          onScroll={[Function]}
                          onScrollBeginDrag={[Function]}
                          onScrollEndDrag={[Function]}
                          removeClippedSubviews={false}
                          renderItem={[Function]}
                          renderScrollComponent={[Function]}
                          scrollEventThrottle={50}
                          stickyHeaderIndices={[]}
                          viewabilityConfigCallbackPairs={[]}
                          waitFor={
                            [
                              {
                                "current": null,
                              },
                              {
                                "current": null,
                              },
                            ]
                          }
                        >
                          <View />
                        </RCTScrollView>
                      </View>
                      <Modal
                        animationType="slide"
                        hardwareAccelerated={false}
                        modifiers={
                          {
                            "alignments": {},
                            "backgroundColor": undefined,
                            "borderRadius": undefined,
                            "color": "transparent",
                            "flexStyle": undefined,
                            "margins": {},
                            "paddings": {},
                            "positionStyle": undefined,
                            "typography": undefined,
                          }
                        }
                        testID="filterModal"
                        transparent={true}
                        visible={false}
                      />
                    </View>
                  </RCTScrollView>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Search"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
